### PR TITLE
fix: support entry as flow in pf flow test --ui

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

Fix the issue that `pf flow test --ui` can't accept entry as flow. 

image

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/_utilities/chat_utils.py [hunk 1]
```diff
@@ -26,6 +29,33 @@ def construct_flow_absolute_path(flow: str) -> str:
     return (flow_dir / flow_file).absolute().resolve().as_posix()
 
 
+@contextlib.contextmanager
+def generate_yaml_entry_under_flow_directory(entry: Union[str, PathLike, Callable], code: Path = None):
```
Comment (YingChen1996) [inline src/promptflow-devkit/promptflow/_sdk/_utilities/chat_utils.py:33]: May I ask what's the difference of generate_yaml_entry and generate_yaml_entry_under_flow_directory. Maybe can use generate_yaml_entry  directly?

Comment (elliotzh) [reply src/promptflow-devkit/promptflow/_sdk/_utilities/chat_utils.py:33]: `generate_yaml_entry` will generate yaml in a temp folder, while we need to generate it under flow directory in current scenario; on the other hand, we should avoid doing this if possible, so I haven't added a parameter to the origin function.

File: src/promptflow-devkit/promptflow/_cli/_pf/_flow.py [hunk 2]
```diff
@@ -522,18 +527,20 @@ def _test_flow_multi_modal(args, pf_client):
...
-            serve_app_port=serve_app_port,
-            pfs_port=pfs_port,
-            url_params=list_of_dict_to_dict(args.url_params),
-            init=list_of_dict_to_dict(args.init),
-            enable_internal_features=enable_internal_features,
-            skip_open_browser=args.skip_open_browser,
-        )
+        # this is a workaround given current serve app requires flow file to be under flow directory
+        # should switch to generate_yaml_entry_without_delete in the future
+        with generate_yaml_entry_under_flow_directory(entry=args.flow) as flow:
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_pf/_flow.py:532]: not sure if we generate the yaml in temp folder like normal flow test do ?

Comment (elliotzh) [reply src/promptflow-devkit/promptflow/_cli/_pf/_flow.py:532]: please check the corresponding error message. the error is caused by the limitation that current serve app doesn't accept flow_file as a parameter. I will prepare a PR for this but thought it's dangerous to do such a change on serve app near release cut

File: src/promptflow-devkit/promptflow/_sdk/_utilities/chat_utils.py (hunk 1 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_utilities/chat_utils.py:33]: `generate_yaml_entry` will generate yaml in cwd(I think it's flow dir in this scenerio) if code is None

File: src/promptflow-devkit/promptflow/_cli/_pf/_flow.py (hunk 2 above)
Comment (elliotzh) [reply src/promptflow-devkit/promptflow/_cli/_pf/_flow.py]: updated the implementation

File: src/promptflow-devkit/promptflow/_sdk/_utilities/chat_utils.py (hunk 1 above)
Comment (elliotzh) [reply src/promptflow-devkit/promptflow/_sdk/_utilities/chat_utils.py]: update the implementation

**Review by wangchao1230 — DISMISSED** [review]

**Review by lalala123123 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**